astropy/table/tests/test_table.py

```diff
@@ -1382,7 +1382,61 @@ def test_disallow_inequality_comparisons():
         t <= -1.1
 
 
-def test_equality():
+def test_cols_equal():
+
+    col1 = [1, 2]
+    col2 = [1.0, 2.0]
+    col3 = ['a', 'b']
+    t1 = table.Table([col1, col2, col3], names=['a', 'b', 'c'])
+    t2 = table.Table([col1, col2], names=['a', 'b'])
+    t3 = table.table_helpers.simple_table()
+    tm = table.Table([col1, col2, col3], names=['a', 'b', 'c'])
+    tm['tm'] = Time([1, 2], format='cxcsec')
+    tm1 = tm
+    tm1['tm'][0] = np.ma.masked
```

Comment:
This will mask the entry in `tm` too.  Also, let's call the column 'time' to avoid confusion.

Reply:
Yup, will fix.